### PR TITLE
add aarch64 support

### PR DESCRIPTION
**Pull request by ausil**
Signed-off-by: Dennis Gilmore <dennis@ausil.us>

### CODE REVIEW

File: xorg-x11-drv-nvidia.spec [hunk 1]
```diff
@@ -205,10 +206,17 @@ Advanced  power management, preserve memory allocation on suspend/resume.
...
 %setup -q -c -T
+%ifarch x86_64
 sh %{SOURCE0} \
   --extract-only --target nvidiapkg-x86_64
 ln -s nvidiapkg-x86_64 nvidiapkg
-
+%else
+%ifarch aarch64
+sh %{SOURCE1} \
+  --extract-only --target nvidiapkg-aarch64
```
Comment (kwizart) [inline xorg-x11-drv-nvidia.spec]: Can we simplify using %{_target_cpu} ?

File: xorg-x11-drv-nvidia.spec [hunk 2]
```diff
@@ -357,11 +373,16 @@ mkdir -p %{buildroot}%{_sysconfdir}/nvidia
 
 #Install the nvidia kernel modules sources archive
 mkdir -p %{buildroot}%{_datadir}/nvidia-kmod-%{version}
+%ifarch x86_64
 tar Jcf %{buildroot}%{_datadir}/nvidia-kmod-%{version}/nvidia-kmod-%{version}-x86_64.tar.xz kernel kernel-open
-
+%else
+tar Jcf %{buildroot}%{_datadir}/nvidia-kmod-%{version}/nvidia-kmod-%{version}-aarch64.tar.xz kernel kernel-open
```
Comment (kwizart) [inline xorg-x11-drv-nvidia.spec]: Same simplification with %{_target_cpu}

File: xorg-x11-drv-nvidia.spec [hunk 3]
```diff
@@ -468,7 +492,7 @@ fi ||:
 %files kmodsrc
 %dir %{_datadir}/nvidia-kmod-%{version}
 %{rpmmacrodir}/macros.%{name}-kmodsrc
-%{_datadir}/nvidia-kmod-%{version}/nvidia-kmod-%{version}-x86_64.tar.xz
+%{_datadir}/nvidia-kmod-%{version}/nvidia-kmod-%{version}-*.tar.xz
```
Comment (kwizart) [inline xorg-x11-drv-nvidia.spec]: Same simplification,but wildcard is fine also.

**Review by kwizart — CHANGES_REQUESTED** [review]
I've this task pending for some time, so it's finally there.
Thanks for working on this !

File: xorg-x11-drv-nvidia.spec (hunk 1 above)
Comment (ausil) [reply xorg-x11-drv-nvidia.spec]: Perhaps, though each arch has a different source, so you have to extract from a different file. you could probably do away with the ln step and just extract both to nvidiapkg

File: xorg-x11-drv-nvidia.spec (hunk 2 above)
Comment (ausil) [reply xorg-x11-drv-nvidia.spec]: I used %{_arch}

File: xorg-x11-drv-nvidia.spec (hunk 3 above)
Comment (ausil) [reply xorg-x11-drv-nvidia.spec]: switched to %{_arch} also

File: xorg-x11-drv-nvidia.spec [hunk 4]
```diff
@@ -534,7 +552,9 @@ fi ||:
 %{_libdir}/libcuda.so.%{version}
 %{_libdir}/libnvcuvid.so.1
 %{_libdir}/libnvcuvid.so.%{version}
+%ifarch x86_64
```
Comment (kwizart) [inline xorg-x11-drv-nvidia.spec]: You are using %ifarch here, but ifnarch aarch64 earlier for the libnvidia-compiler.so.%{version} case
It seems better to assume %ifnarch aarch64 for both cases (to handle i686 libnvidia-compiler.so.%{version} that still exist).

Comment (ausil) [reply xorg-x11-drv-nvidia.spec]: yep, fixed.

**Comment by kwizart** [comment]
Merged (with few minor changes).

**Pull request closed without merging**